src/runtime-rs/crates/resource/src/volume/mod.rs

```diff
@@ -0,0 +1,106 @@
+// Copyright (c) 2019-2022 Alibaba Cloud
+// Copyright (c) 2019-2022 Ant Group
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+mod block_volume;
+mod default_volume;
+mod share_fs_volume;
+mod shm_volume;
+
+use std::{sync::Arc, vec::Vec};
+
+use anyhow::{Context, Result};
+use tokio::sync::RwLock;
+
+use crate::share_fs::ShareFs;
+
+pub trait Volume: Send + Sync {
+    fn get_volume_mount(&self) -> Result<Vec<oci::Mount>>;
+    fn get_storage(&self) -> Result<Vec<agent::Storage>>;
+    fn cleanup(&self) -> Result<()>;
+}
+
+#[derive(Default)]
+pub struct VolumeResourceInner {
+    volumes: Vec<Arc<dyn Volume>>,
+}
+
+impl VolumeResourceInner {
+    fn new() -> Self {
+        Self { volumes: vec![] }
+    }
+}
+
+#[derive(Default)]
+pub struct VolumeResource {
+    inner: Arc<RwLock<VolumeResourceInner>>,
+}
+
+impl VolumeResource {
+    pub fn new() -> Self {
+        Self {
```

Comment:
It's better to use Default::default()

Reply:
Thanks for your review, change to use `Default::default()`.